### ISSUE TITLE
{"error":false,"message":"Unrecognized path"}

### ISSUE THREAD

**Issue by matjazpotocnik** [Toutouwai/ProcessSessionInfo#2]
I get this, will check later what could be the reason... 

Most likely it's coming from:

PHP Notice: file_get_contents(): Read of 29093 bytes failed with errno=13 Permission denied in \wire\core\WireFileTools.php:1170

Why? Don't know, maybe OS is writing to the file at the same time?

**Comment by matjazpotocnik** [comment]
It turns out that a current session file is locked and can't be read (tried different methods, no luck). The solution is to copy a file somewhere and then read it. Something like this:

```php
  private function ff_get($f) {
    $handle = @fopen($f, "r");
    $stream = stream_get_meta_data($handle);
    if($stream['blocked']) {
      @fclose($handle);
      $tf = (string) wireTempDir('SessionViewer') . basename($f);
      @wireCopy($f, $tf);
      $f = $tf;
      $handle = @fopen($f, "r");
      if($handle === false) return false;
    }
    $contents = @fread($handle, @filesize($f));
    @fclose($handle);
    @unlink($tf);
    if($contents) return $contents;
    return false;
  }
```

Also: https://processwire.com/modules/session-viewer/

**Comment by Toutouwai** [comment]
@matjazpotocnik, I'm pretty sure that the session file that gives that error is your own session. No info for that session is needed in the module output (because you already know that you yourself are using the website) so the intention is that when a locked file is encountered that prevents the contents being read this session gets skipped and the loop continues with the next file.

That is the purpose of these lines: https://github.com/Toutouwai/ProcessSessionInfo/blob/c90d8968fa402736c3241cabce558e45ccecad28/ProcessSessionInfo.module.php#L68-L71C5

I don't want to do anything like make a copy of each session file because the module needs to loop over up to 500 files and I think the overhead will be too great.

The `@` operator before `$files->fileGetContents($filepath)` is supposed to suppress error notices, so I'm not sure why you are seeing the error. Where does the `{"error":false,"message":"Unrecognized path"}` message appear?

And is there anything unusual about your environment that would prevent the `@` operator from working? Since PHP 8 the `@` operator won't suppress fatal errors but the "errno=13 Permission denied" should just be a notice and not fatal. I'm testing on PHP 8.2 here and I can see the error when the `@` operator isn't used, but when it is used it suppresses the error as expected.

Could you try changing [this line](https://github.com/Toutouwai/ProcessSessionInfo/blob/c90d8968fa402736c3241cabce558e45ccecad28/ProcessSessionInfo.module.php#L67) to...
```php
$contents = @file_get_contents($filepath);
```
...and see if that makes a difference?

**Comment by matjazpotocnik** [comment]
> I'm pretty sure that the session file that gives that error is your own session. 

Yes, that's what I ment with: "current session file is locked and can't be read".

> No info for that session is needed in the module output (because you already know that you yourself are using the website)

When I installed a module, I expected to see my session. It wasn't listed, so I started looking for the reason. I prefer to see my own session, which is visually different from other sessions.

`I don't want to do anything like make a copy of each session file because the module needs to loop over up to 500 files and I think the overhead will be too great.`

Of course, copy would only be needed for you own session (or a session that is locked), not for every session file.

> "errno=13 Permission denied" should just be a notice and not fatal

Yes, it's a notice, and @ suppress works. I removed it to see if I could find out where the original error is coming from - namely `{"error":false,"message":"Unrecognized path"}` which was returned from the ayax call. Oddly, I can't replicate it any more.
